index.js

```diff
@@ -39,23 +41,37 @@ module.exports = {
     }
 
     if (debugMode) {
-      console.warn('debugMode is not implemented yet for this plugin')
+      console.warn(`${chalk.yellow(
+        '@netlify/plugin-algolia-index:'
+      )} ${chalk.blueBright('debugMode')} is not implemented yet for this plugin`)
     }
 
-    let searchIndex = []
+    // Walk publish directory for files to parse
     const newManifest = await walk(PUBLISH_DIR, exclude)
 
+    // Parse all files for their indexable content
+    let newIndex = []
     await Promise.all(
       newManifest.map(async (htmlFilePath) => {
         const htmlFileContent = await readFile(htmlFilePath, 'utf8')
-        searchIndex.push(await parse(htmlFileContent, htmlFilePath, { PUBLISH_DIR, textLength, stopwords }))
+        newIndex.push(await parse(htmlFileContent, htmlFilePath, { PUBLISH_DIR, textLength, stopwords }))
       })
     )
 
-    // export content to algolia
-    await exporter(searchIndex)
+    // Export content to Algolia
+    try {
+      const client = algoliasearch(algoliaAppId, algoliaAdminKey)
+      const index = client.initIndex(algoliaIndex)
+      await exporter(index, newIndex)
+    } catch (error) {
+      console.error(error)
```

Comment:
When using `build.fail*()`, the error will already be printed in logs, so I think this `console.error()` might be redundant.

Reply:
Ahh good point :)